x-pack/plugins/index_lifecycle_management/public/application/services/policies/cold_phase.ts

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { isEmpty } from 'lodash';
+import {
+  AllocateAction,
+  ColdPhase,
+  SerializedColdPhase,
+  serializedPhaseInitialization,
+} from './types';
+import { isNumber, splitSizeAndUnits } from './policy_serialization';
+import {
+  numberRequiredMessage,
+  PhaseValidationErrors,
+  positiveNumberRequiredMessage,
+} from './policy_validation';
+
+const coldPhaseInitialization: ColdPhase = {
+  phaseEnabled: false,
+  selectedMinimumAge: '0',
+  selectedMinimumAgeUnits: 'd',
+  selectedNodeAttrs: '',
+  selectedReplicaCount: '',
+  freezeEnabled: false,
+  phaseIndexPriority: '',
+};
+
+export const coldPhaseFromES = (phaseSerializedOrUndefined?: SerializedColdPhase): ColdPhase => {
+  const phase = { ...coldPhaseInitialization };
+  if (!phaseSerializedOrUndefined) {
+    return phase;
+  }
+
+  // after ! check it's safe to cast type as a serialized phase
```

Comment:
Is this comment still relevant?
I do think it is strange (1) to give it a name with `OrUndefined` in it  and (2) to cast it.

It would be safer to have a logic if it is undefined and not add the suffix.

Something like this

```js
if (phaseSerialized === undefined){
  // explicitly deal with this case
  return null;
}

```

Reply:
I agree with this suggestion, so I renamed the parameter and added an `undefined` and `null` check.